### PR TITLE
Update the guides to refer to the io.quarkus.platform BOMs instead of the universe BOM

### PR DESCRIPTION
**Pull request by aloubyansky**

### CODE REVIEW

File: docs/src/main/asciidoc/building-my-first-extension.adoc [hunk 1]
```diff
@@ -767,25 +767,15 @@ That's why there is the Quarkus Platform.
...
-If the extension solves a general problem, it is very handy for Quarkus users to see it on https://code.quarkus.io.
-But this comes with some responsibility for you, keeping it up to date with Quarkus minor releases (every month or so at the moment).
-When in doubt, have a conversation with the community in the https://groups.google.com/forum/#!forum/quarkus-dev[Quarkus Google Group]. We can make a collective decision.
-
-If you want to contribute a new extension, we strongly suggest you to host it in the Quarkiverse Hub.
+Quarkus platform is a set of extensions that target the primary use-cases of Quarkus as a development framework and can safely be used in any combination in the same application without creating a dependency conflict.
+From an application developer perspective, a Quarkus platform is represented as one or more Maven BOMs, for example `io.quarkus.platform:quarkus-bom:{quarkus-version}`, `io.quarkus.platform:quarkus-kogito-bom:{quarkus-version}`, etc, whose dependency version constraints were aligned globally aligned so that these BOMs can be imported in the same application in any order w/o introducing a dependency conflict.
```
Comment (gastaldi) [inline docs/src/main/asciidoc/building-my-first-extension.adoc:771]:
```suggestion
From an application developer perspective, a Quarkus platform is represented as one or more Maven BOMs, for example `io.quarkus.platform:quarkus-bom:{quarkus-version}`, `io.quarkus.platform:quarkus-kogito-bom:{quarkus-version}`, etc, whose dependency version constraints were globally aligned so that these BOMs can be imported in the same application in any order w/o introducing a dependency conflict.
```

File: docs/src/main/asciidoc/building-my-first-extension.adoc [hunk 2]
```diff
@@ -767,25 +767,15 @@ That's why there is the Quarkus Platform.
...
 
 *Quarkiverse Hub*
 
 link:https://github.com/quarkiverse[Quarkiverse Hub] is the GitHub organization that provides repository hosting (including build, CI and release publishing setup) for Quarkus extension projects contributed by the community.
 
+In case you are wondering about creating a new Quarkus extension and adding it to the Quarkus ecosystem that the Quarkus community can disover using the Quarkus dev tools (including the https://quarkus.io/guides/cli-tooling[Quarkus CLI] and https://code.quarkus.io[code.quarkus.io]), the https://github.com/quarkiverse[Quarkiverse Hub] GitHub organization will be a good home for it.
```
Comment (gastaldi) [inline docs/src/main/asciidoc/building-my-first-extension.adoc:777]:
```suggestion
In case you are wondering about creating a new Quarkus extension and adding it to the Quarkus ecosystem that the Quarkus community can discover using the Quarkus dev tools (including the https://quarkus.io/guides/cli-tooling[Quarkus CLI] and https://code.quarkus.io[code.quarkus.io]), the https://github.com/quarkiverse[Quarkiverse Hub] GitHub organization will be a good home for it.
```

File: docs/src/main/asciidoc/building-my-first-extension.adoc [hunk 3]
```diff
@@ -795,6 +785,8 @@ We'll take care of provisioning a new repository and set it up to:
 - Configure your extension to use the link:https://github.com/quarkusio/quarkus-ecosystem-ci#quarkus-ecosystem-ci[Quarkus Ecosystem CI] to build against the latest Quarkus Core changes;
 - Give you the freedom to manage the project and release to Maven Central as you like.
 
+NOTE: Extensions hosted in the Quarkiverse Hub may or may not end up in the Quarkus platform.
```
Comment (gastaldi) [inline docs/src/main/asciidoc/building-my-first-extension.adoc:788]: Is there any process someone could follow to have a Quarkiverse extension added to the platform?

Comment (aloubyansky) [reply docs/src/main/asciidoc/building-my-first-extension.adoc:788]: This would have to be discussed with the platform maintainers. Specifically, the extension should fit the primary purposes/use-cases the platform is targeting and match the quality requirements, i.e. be supersonic, subatomic and work in native.

Comment (gastaldi) [reply docs/src/main/asciidoc/building-my-first-extension.adoc:788]: Right, just wondering if that would be something that readers might ask frequently, maybe we should elaborate more this note at some point (doesn't need to be in this PR, ofc)

File: docs/src/main/asciidoc/building-my-first-extension.adoc [hunk 4]
```diff
@@ -767,25 +767,15 @@ That's why there is the Quarkus Platform.
...
-If the extension solves a general problem, it is very handy for Quarkus users to see it on https://code.quarkus.io.
-But this comes with some responsibility for you, keeping it up to date with Quarkus minor releases (every month or so at the moment).
-When in doubt, have a conversation with the community in the https://groups.google.com/forum/#!forum/quarkus-dev[Quarkus Google Group]. We can make a collective decision.
-
-If you want to contribute a new extension, we strongly suggest you to host it in the Quarkiverse Hub.
+Quarkus platform is a set of extensions that target the primary use-cases of Quarkus as a development framework and can safely be used in any combination in the same application without creating a dependency conflict.
+From an application developer perspective, a Quarkus platform is represented as one or more Maven BOMs, for example `io.quarkus.platform:quarkus-bom:{quarkus-version}`, `io.quarkus.platform:quarkus-kogito-bom:{quarkus-version}`, etc, whose dependency version constraints were globally aligned so that these BOMs can be imported in the same application in any order w/o introducing a dependency conflict.
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-my-first-extension.adoc:771]:
```suggestion
From an application developer perspective, a Quarkus platform is represented as one or more Maven BOMs, for example `io.quarkus.platform:quarkus-bom:{quarkus-version}`, `io.quarkus.platform:quarkus-kogito-bom:{quarkus-version}`, etc, whose dependency version constraints were globally aligned so that these BOMs can be imported in the same application in any order without introducing a dependency conflict.
```

File: docs/src/main/asciidoc/building-my-first-extension.adoc [hunk 5]
```diff
@@ -767,25 +767,15 @@ That's why there is the Quarkus Platform.
...
 
 *Quarkiverse Hub*
 
 link:https://github.com/quarkiverse[Quarkiverse Hub] is the GitHub organization that provides repository hosting (including build, CI and release publishing setup) for Quarkus extension projects contributed by the community.
 
+In case you are wondering about creating a new Quarkus extension and adding it to the Quarkus ecosystem that the Quarkus community can discover using the Quarkus dev tools (including the https://quarkus.io/guides/cli-tooling[Quarkus CLI] and https://code.quarkus.io[code.quarkus.io]), the https://github.com/quarkiverse[Quarkiverse Hub] GitHub organization will be a good home for it.
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-my-first-extension.adoc:777]:
```suggestion
In case you are wondering about creating a new Quarkus extension and adding it to the Quarkus ecosystem so that the Quarkus community can discover it using the Quarkus dev tools (including the https://quarkus.io/guides/cli-tooling[Quarkus CLI] and https://code.quarkus.io[code.quarkus.io]), the https://github.com/quarkiverse[Quarkiverse Hub] GitHub organization will be a good home for it.
```
would be better? Otherwise it's hard to follow the sentence.

File: docs/src/main/asciidoc/platform.adoc [hunk 6]
```diff
@@ -157,26 +157,26 @@ IMPORTANT: The content of the properties artifacts found earlier will dominate o
...
-        <groupId>io.quarkus</groupId>
-        <artifactId>quarkus-universe-bom-quarkus-platform-properties</artifactId>
+        <groupId>org.acme</groupId>
+        <artifactId>acme-quarkus-bom-quarkus-platform-properties</artifactId>
         <type>properties</type>
         <version>${project.version}</version>
       </dependency>
 
-      <!-- Quarkus Core BOM -->
+      <!-- Quarkus platform BOM -->
```
Comment (gsmet) [inline docs/src/main/asciidoc/platform.adoc:179]: Is it the platform BOM given you kept it `io.quarkus`?

File: docs/src/main/asciidoc/platform.adoc [hunk 7]
```diff
@@ -188,4 +188,4 @@ For example, the `quarkus-universe-bom` platform is based on the `quarkus-bom` p
   <!-- skipped content -->
 ----
 
-That way, the `quarkus-universe-bom` platform properties will appear before the `quarkus-bom` platform properties and so will be dominating during merging.
+That way, the `org.acme:acme-quarkus-bom` platform properties will appear before the `io.quarkus:quarkus-bom` platform properties and so will be dominating at build time.
```
Comment (gsmet) [inline docs/src/main/asciidoc/platform.adoc:191]: Same here, did you want to keep it `io.quarkus`?

**Review by gsmet — CHANGES_REQUESTED** [review]
I spotted a few things.

Could you squash everything when done? Thanks!

File: docs/src/main/asciidoc/platform.adoc (hunk 6 above)
Comment (aloubyansky) [reply docs/src/main/asciidoc/platform.adoc:179]: Technically, `io.quarkus` is a platform but yes, I think it'd be better to change it to `io.quarkus.platform`.

**Comment by aloubyansky** [comment]
I decided not to change the last example to `io.quarkus.platform`, because I am not sure it's what generally should be used as a base for a custom platform.

**Review by gsmet — APPROVED** [review]
